stablehlo/reference/Ops.cpp

```diff
@@ -76,6 +76,97 @@ Tensor eval(CosineOp op, const Tensor &operand) {
   return result;
 }
 
+Tensor eval(DotGeneralOp op, const Tensor &lhs, const Tensor &rhs) {
+  Tensor result(op.getType());
+  ArrayRef<int64_t> lhsShape = lhs.getType().getShape();
+  ArrayRef<int64_t> lhsBatchingDims =
+      op.getDotDimensionNumbers().getLhsBatchingDimensions();
+  ArrayRef<int64_t> rhsBatchingDims =
+      op.getDotDimensionNumbers().getRhsBatchingDimensions();
+  ArrayRef<int64_t> lhsContractingDims =
+      op.getDotDimensionNumbers().getLhsContractingDimensions();
+  ArrayRef<int64_t> rhsContractingDims =
+      op.getDotDimensionNumbers().getRhsContractingDimensions();
+  std::vector<int64_t> lhsNonContractingDims;
```

Comment:
rename it to lhsNonBatchingNonContractingDims? 
Same with rhsNonContractingDims?